server/src/test/java/org/elasticsearch/index/mapper/SyntheticSourceNativeArrayIntegrationTests.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.util.BitUtil;
+import org.elasticsearch.action.admin.indices.forcemerge.ForceMergeRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.common.io.stream.ByteArrayStreamInput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.query.IdsQueryBuilder;
+import org.elasticsearch.test.ESSingleNodeTestCase;
+import org.hamcrest.Matchers;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+public class SyntheticSourceNativeArrayIntegrationTests extends ESSingleNodeTestCase {
+
+    public void testSynthesizeArray() throws Exception {
+        var arrayValues = new Object[][] {
+            new Object[] { "z", "y", null, "x", null, "v" },
+            new Object[] { null, "b", null, "a" },
+            new Object[] { null },
+            new Object[] { null, null, null },
+            new Object[] { "c", "b", "a" } };
+        verifySyntheticArray(arrayValues);
+    }
+
+    public void testSynthesizeEmptyArray() throws Exception {
+        var arrayValues = new Object[][] { new Object[] {} };
+        verifySyntheticArray(arrayValues);
+    }
+
+    public void testSynthesizeArrayRandom() throws Exception {
```

Comment:
Do we test with `ignore_above` anywhere?

Reply:
Added in `SyntheticSourceNativeArrayIntegrationTests#testSynthesizeArrayIgnoreAbove(...)`